shiny/ui/_markdown_stream.py

```diff
@@ -0,0 +1,276 @@
+from contextlib import asynccontextmanager
+from typing import AsyncIterable, Iterable, Literal, Union
+
+from htmltools import css
+
+from .. import _utils, reactive
+from .._docstring import add_example
+from .._typing_extensions import TypedDict
+from ..session import require_active_session
+from ..types import NotifyException
+from ..ui.css import CssUnit, as_css_unit
+from . import Tag
+from ._html_deps_py_shiny import markdown_stream_dependency
+
+__all__ = (
+    "markdown_stream_ui",
+    "MarkdownStream",
+)
+
+StreamingContentType = Literal[
+    "markdown",
+    "html",
+    "semi-markdown",
+    "text",
+]
+
+
+class ContentMessage(TypedDict):
+    id: str
+    content: str
+    operation: Literal["append", "replace"]
+
+
+class isStreamingMessage(TypedDict):
+    id: str
+    isStreaming: bool
+
+
+@add_example()
+class MarkdownStream:
+    """
+    Stream markdown (or HTML) content.
```

Comment:
```suggestion
    A component for streaming markdown or HTML content.
```

Trying to avoid noun/verb confusion around "stream". To me, a class describes a _noun_ but _stream_ here is used as a verb.